dashboard/src/components/partials/EntitiesList/EntitiesListHeader.js

```diff
@@ -31,18 +34,36 @@ const styles = theme => ({
 
 class EntitiesListHeader extends React.PureComponent {
   static propTypes = {
+    classes: PropTypes.object.isRequired,
     onClickSelect: PropTypes.func,
+    onChangeFilter: PropTypes.func,
     selectedCount: PropTypes.number,
-    classes: PropTypes.object.isRequired,
+    subscriptions: PropTypes.shape({ entries: PropTypes.array }),
   };
 
   static defaultProps = {
     onClickSelect: () => {},
+    onChangeFilter: () => {},
     selectedCount: 0,
+    subscriptions: { entries: [] },
+  };
+
+  static fragments = {
+    subscriptions: gql`
+      fragment EntitiesListHeader_subscriptions on SubscriptionSet {
+        entries(orderBy: FREQUENCY)
```

Comment:
Does this run the risk of returning a dangerously large result without a limit?
 	 
I think making this it's own query would be a good idea. With query batching there wouldn't be any overhead, and it would set us up for fetching this list in chunks if we need to or refetching it independently of the main page content.

Reply:
> Does this run the risk of returning a dangerously large result without a limit?

From looking around the set could probably be in the hundreds, so it is probably worth adding a limit (and offset.) 👍

> I think making this it's own query would be a good idea.

I like your thinking here, with a limit implemented, it would make it easy to implement react virtualized and friends. However, with the design in flux, I think I'll leave it as is for the time being. Lest I build a more involved solution all for naught.

Reply:
cool cool cool